plone/app/tiles/browser/download.py

```diff
@@ -87,3 +94,10 @@ class DisplayFile(NamedfileDisplayFile):
 
     def _getFile(self):
         return _shared_getFile(self)
+
+    def get_canonical(self, file):
+        filename = getattr(file, "filename", None)
+        if filename is None:
+            return f"{self.context.url}/@@display-file/{self.fieldname}"
```

Comment:
ok, next question :) @petschki
are you shure this works, when self.url returns an url including query parameters?
also does traversing work with multiple views in between? /some/object/@@tile/@@download? it would probably, if the @@tile returns an traversable object or implements a traverser itself...

Reply:
Super about the traversing. double view is unusal, but sure, it should work if the first view returns something taversable itself.

Regarding `absolute_url` - yes I agree. We should implement absolute url in persistent title.